src/signal.rs

```diff
@@ -1903,4 +1928,170 @@ mod menu_item {
+        }
+    }
+
+    unsafe extern "C" fn add_trampoline<T>(this: *mut GtkCellArea,
+                                           renderer: *mut GtkCellRenderer,
+                                           editable: *mut GtkCellEditable,
+                                           cell_area: *mut GdkRectangle,
+                                           path: *const c_char,
+                                           f: &Box<Fn(&T, &CellRenderer, &CellEditable, &Rectangle, &str) + 'static>)
+    where T: IsA<CellArea> {
+        callback_guard!();
+        let buf = CStr::from_ptr(path).to_bytes();
+        let string = str::from_utf8(buf).unwrap();
+        f(&CellArea::from_glib_none(this).downcast_unchecked(),
+          &CellRenderer::from_glib_none(renderer),
+          &CellEditable::from_glib_none(editable),
+          &Rectangle { x: (*cell_area).x, y: (*cell_area).y, width: (*cell_area).width, height: (*cell_area).height },
+          string);
+    }
+
+    unsafe extern "C" fn apply_trampoline<T>(this: *mut GtkCellArea,
+                                             model: *mut GtkTreeModel,
+                                             iter: *mut GtkTreeIter,
+                                             is_expander: gboolean,
+                                             is_expanded: gboolean,
+                                             f: &Box<Fn(&T, &TreeModel, &TreeIter, bool, bool) + 'static>)
+    where T: IsA<CellArea> {
+        callback_guard!();
+        f(&CellArea::from_glib_none(this).downcast_unchecked(),
+          &TreeModel::from_glib_none(model).downcast_unchecked(),
+          &TreeIter::from_glib_none(iter),
+          from_glib(is_expander),
+          from_glib(is_expanded));
+    }
+
+    unsafe extern "C" fn focus_trampoline<T>(this: *mut GtkCellArea,
+                                             renderer: *mut GtkCellRenderer,
+                                             path: *const c_char,
+                                             f: &Box<Fn(&T, &CellRenderer, &str) + 'static>)
+    where T: IsA<CellArea> {
+        callback_guard!();
+        let buf = CStr::from_ptr(path).to_bytes();
+        let string = str::from_utf8(buf).unwrap();
+        f(&CellArea::from_glib_none(this).downcast_unchecked(),
+          &CellRenderer::from_glib_none(renderer).downcast_unchecked(),
+          string);
+    }
+
+    unsafe extern "C" fn remove_trampoline<T>(this: *mut GtkCellArea,
+                                              renderer: *mut GtkCellRenderer,
+                                              editable: *mut GtkCellEditable,
+                                              f: &Box<Fn(&T, &CellRenderer, &CellEditable) + 'static>)
+    where T: IsA<CellArea> {
+        callback_guard!();
+        f(&CellArea::from_glib_none(this).downcast_unchecked(),
+          &CellRenderer::from_glib_none(renderer).downcast_unchecked(),
+          &CellEditable::from_glib_none(editable).downcast_unchecked());
```

Comment:
Just generally it only makes sense to downcast `this` because the rest of arguments  aren't specializable.